### PR TITLE
Added validations for the join us form

### PR DESCRIPTION
**Pull request by Jassi10000**
## Issue that this pull request solves

 Closes: #506  (issue number)

## Proposed changes
Added validations checks for : 
 
Name
Email
Interested Domain
Year of Study
College Name

## Types of changes

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (Documentation content changed)
- [x] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface

snap1

## Other information

Any other information that is important to this pull request

### CODE REVIEW

File: frontend/src/pages/Home/components/JoinUsForm/Form.jsx [hunk 1]
```diff
@@ -49,16 +174,26 @@ export const JoinUsForm = (props) => {
                   }
                 >
                   <input
-                    placeholder="Name"
-                    id="txt_name"
+                    placeholder="Email ID"
+                    id="txt_email"
                     type="text"
                     required="required"
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Home/components/JoinUsForm/Form.jsx:180]: as we are doing the validation using joi so I believe we don't need these inbuilt validations

File: frontend/src/pages/Home/components/JoinUsForm/Form.jsx [hunk 2]
```diff
@@ -67,311 +202,318 @@ export const JoinUsForm = (props) => {
                   }
                 >
                   <input
-                    placeholder="Contact No."
-                    id="phone"
-                    type="tel"
+                    placeholder="Linkedin Profile URL"
+                    id="txt_link"
+                    type="text"
                     required="required"
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Home/components/JoinUsForm/Form.jsx:208]: same as above(no need)

File: frontend/src/pages/Home/components/JoinUsForm/Form.jsx [hunk 3]
```diff
@@ -67,311 +202,318 @@ export const JoinUsForm = (props) => {
...
-                  placeholder="Email ID"
-                  id="txt_email"
-                  type="text"
+                <textarea
+                  placeholder="How can you contribute to help the community?"
+                  id="txt_desc"
+                  rows="2"
+                  cols="20"
+                  name="desc"
                   required="required"
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Home/components/JoinUsForm/Form.jsx:235]: same(no need)

File: frontend/package.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     "bootstrap": "^4.5.3",
     "bootstrap-social": "^5.1.1",
     "font-awesome": "^4.7.0",
+    "joi": "^17.4.0",
```
Comment (Kajol-Kumari) [inline frontend/package.json:15]: I believe we don't need it. Can you please remove it?

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @Jassi10000 this PR needs minor enhancements, please incorporate these requested changes
1. The error message doesn't appear if I directly click on the `submit` button.It's appearing only when I start typing in the input box. Please fix it.
2. The error message should appear near the respective input box(please see other forms for the reference of how the error message is appearing for them)

File: frontend/package.json (hunk 4 above)
Comment (Jassi10000) [reply frontend/package.json:15]: Sure

File: frontend/src/pages/Home/components/JoinUsForm/Form.jsx (hunk 2 above)
Comment (Jassi10000) [reply frontend/src/pages/Home/components/JoinUsForm/Form.jsx:208]: actually these were already present in the file before I did anything

Anyways , I will remove them , no issues

File: frontend/src/pages/Home/components/JoinUsForm/Form.jsx (hunk 1 above)
Comment (Jassi10000) [reply frontend/src/pages/Home/components/JoinUsForm/Form.jsx:180]: actually these were already present in the file before I did anything

Anyways , I will remove them , no issues

**Comment by Jassi10000** [comment]
> Hey @Jassi10000 this PR needs minor enhancements, please incorporate these requested changes
> 
> 1. The error message doesn't appear if I directly click on the `submit` button.It's appearing only when I start typing in the input box. Please fix it.
> 2. The error message should appear near the respective input box(please see other forms for the reference of how the error message is appearing for them)

You mean to say that if someone is submitting an empty form , so I need to return an error , am I right in understanding your requested changes ?

**Comment by Kajol-Kumari** [comment]
> > Hey @Jassi10000 this PR needs minor enhancements, please incorporate these requested changes
> > 
> > 1. The error message doesn't appear if I directly click on the `submit` button.It's appearing only when I start typing in the input box. Please fix it.
> > 2. The error message should appear near the respective input box(please see other forms for the reference of how the error message is appearing for them)
> 
> You mean to say that if someone is submitting an empty form , so I need to return an error , am I right in understanding your requested changes ?

Yeah, please see the gif I have attached below
validation_gif

**Comment by Jassi10000** [comment]
> > > Hey @Jassi10000 this PR needs minor enhancements, please incorporate these requested changes
> > > 
> > > 1. The error message doesn't appear if I directly click on the `submit` button.It's appearing only when I start typing in the input box. Please fix it.
> > > 2. The error message should appear near the respective input box(please see other forms for the reference of how the error message is appearing for them)
> > 
> > 
> > You mean to say that if someone is submitting an empty form , so I need to return an error , am I right in understanding your requested changes ?
> 
> Yeah, please see the gif I have attached below
> validation_gif

Ohh okay , I got it ,
will  make the required changes and deliever the correct PR very soon

**Pull request closed without merging**